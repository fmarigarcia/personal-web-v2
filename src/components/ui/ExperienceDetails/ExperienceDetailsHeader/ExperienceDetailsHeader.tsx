import React from 'react';
import { useTranslation } from 'react-i18next';

interface ExperienceDetailsHeaderProps {
    company: string;
    position: string;
    duration: string;
}

export const ExperienceDetailsHeader: React.FC<ExperienceDetailsHeaderProps> = ({
    company,
    position,
    duration,
}) => {
    const { t } = useTranslation();

    return (
        <div className="mb-6 border-b border-gray-100 pb-6">
            <h3 className="text-2xl lg:text-3xl font-bold text-gray-900 mb-2">
                {company}
            </h3>
            <p className="text-blue-600 font-semibold text-lg lg:text-xl mb-2">
                {t(position)}
            </p>
            <p className="text-gray-500 font-medium">
                {t(duration)}
            </p>
        </div>
    );
};
