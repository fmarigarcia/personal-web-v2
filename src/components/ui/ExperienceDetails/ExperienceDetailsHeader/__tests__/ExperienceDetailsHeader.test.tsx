import { render, screen } from '@testing-library/react';
import { ExperienceDetailsHeader } from '../ExperienceDetailsHeader';

// Mock react-i18next
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((key: string) => key),
    }),
}));

describe('ExperienceDetailsHeader', () => {
    const defaultProps = {
        company: 'Test Company',
        position: 'test.position.key',
        duration: 'test.duration.key',
    };

    it('should render company name, position and duration', () => {
        render(<ExperienceDetailsHeader {...defaultProps} />);

        expect(screen.getByText('Test Company')).toBeInTheDocument();
        expect(screen.getByText('test.position.key')).toBeInTheDocument();
        expect(screen.getByText('test.duration.key')).toBeInTheDocument();
    });

    it('should have correct heading structure', () => {
        render(<ExperienceDetailsHeader {...defaultProps} />);

        const heading = screen.getByRole('heading', { level: 3 });
        expect(heading).toHaveTextContent('Test Company');
        expect(heading).toHaveClass(
            'text-2xl',
            'lg:text-3xl',
            'font-bold',
            'text-gray-900',
            'mb-2'
        );
    });

    it('should style position correctly', () => {
        render(<ExperienceDetailsHeader {...defaultProps} />);

        const position = screen.getByText('test.position.key');
        expect(position).toHaveClass(
            'text-blue-600',
            'font-semibold',
            'text-lg',
            'lg:text-xl',
            'mb-2'
        );
    });

    it('should style duration correctly', () => {
        render(<ExperienceDetailsHeader {...defaultProps} />);

        const duration = screen.getByText('test.duration.key');
        expect(duration).toHaveClass('text-gray-500', 'font-medium');
    });

    it('should have correct wrapper structure', () => {
        const { container } = render(
            <ExperienceDetailsHeader {...defaultProps} />
        );

        const wrapper = container.firstChild as HTMLElement;
        expect(wrapper).toHaveClass(
            'mb-6',
            'border-b',
            'border-gray-100',
            'pb-6'
        );
    });

    it('should translate position and duration keys', () => {
        render(
            <ExperienceDetailsHeader
                {...defaultProps}
                position="experience.exp1.position"
                duration="experience.exp1.duration"
            />
        );

        expect(
            screen.getByText('experience.exp1.position')
        ).toBeInTheDocument();
        expect(
            screen.getByText('experience.exp1.duration')
        ).toBeInTheDocument();
    });

    it('should handle long company names', () => {
        render(
            <ExperienceDetailsHeader
                {...defaultProps}
                company="Very Long Company Name That Should Still Display Correctly"
            />
        );

        expect(
            screen.getByText(
                'Very Long Company Name That Should Still Display Correctly'
            )
        ).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <ExperienceDetailsHeader {...defaultProps} />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
