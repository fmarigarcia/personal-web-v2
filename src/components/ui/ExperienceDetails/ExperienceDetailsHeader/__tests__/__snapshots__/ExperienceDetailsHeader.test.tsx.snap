// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceDetailsHeader should match snapshot 1`] = `
<div
  class="mb-6 border-b border-gray-100 pb-6"
>
  <h3
    class="text-2xl lg:text-3xl font-bold text-gray-900 mb-2"
  >
    Test Company
  </h3>
  <p
    class="text-blue-600 font-semibold text-lg lg:text-xl mb-2"
  >
    test.position.key
  </p>
  <p
    class="text-gray-500 font-medium"
  >
    test.duration.key
  </p>
</div>
`;
